File: leetcode/practice/795-number-of-subarrays-with-bounded-maximum/sliding-window.py
class Solution:
    def numSubarrayBoundedMax(self, arr: List[int], lower: int, upper: int) -> int:
        left = right = ans = subArr = 0
        while(right < len(arr)) :
            if arr[right] >= lower and arr[right] <= upper :
                subArr = right - left + 1
            elif arr[right] > upper :
                subArr = 0
                left = right + 1
            
            ans += subArr
            right += 1
        return ans

File: leetcode/practice/795-number-of-subarrays-with-bounded-maximum/sliding-window.cpp
class Solution {
public:
    int numSubarrayBoundedMax(vector<int>& arr , int lowerLim , int upperLim) ;
};

int Solution :: numSubarrayBoundedMax(vector<int>& arr , int lower , int upper)
{
    int i = 0 , j = 0 , ans = 0 , subarr = 0 ;
    while(j < arr.size())
    {
        if(arr[j] >= lower and arr[j] <= upper)
        {
            subarr = j - i + 1 ;
        }
        else if(arr[j] > upper)
        {
            subarr = 0 ;
            i = j + 1 ;
            
        }
        ans += subarr ;
        j++ ;
    }
    return ans ;
}
